contracts/src/access/ownable_two_step.rs

```diff
@@ -0,0 +1,290 @@
+
+    /// Emitted when ownership gets transferred between accounts.
+    event OwnershipTransferred(address indexed previous_owner, address indexed new_owner);
+}
+
+/// An error that occurred in the implementation of an [`Ownable2Step`]
+/// contract.
+#[derive(SolidityError, Debug)]
+pub enum Error {
+    /// Error type from [`Ownable`] contract.
+    Ownable(OwnableError),
+}
+
+sol_storage! {
+    /// State of an `Ownable2Step` contract.
+    pub struct Ownable2Step {
+        /// [`Ownable`] contract.
+        Ownable _ownable;
+        /// Pending owner of the contract.
+        address _pending_owner;
+    }
+}
+
+#[public]
+impl Ownable2Step {
+    /// Returns the address of the current owner.
+    pub fn owner(&self) -> Address {
+        self._ownable.owner()
+    }
+
+    /// Returns the address of the pending owner.
+    pub fn pending_owner(&self) -> Address {
+        self._pending_owner.get()
+    }
+
+    /// Initiates the transfer of ownership to a new account (`new_owner`).
+    /// Can only be called by the current owner.
+    ///
+    /// # Arguments
+    ///
+    /// * `&mut self` - Write access to the contract's state.
+    /// * `new_owner` - The next owner of this contract.
+    ///
+    /// # Errors
+    ///
+    /// If called by any account other than the owner, then the error
+    /// [`OwnableError::UnauthorizedAccount`] is returned.
+    pub fn transfer_ownership(
+        &mut self,
+        new_owner: Address,
+    ) -> Result<(), Error> {
+        self._ownable.only_owner()?;
+        self._pending_owner.set(new_owner);
+
+        let current_owner = self.owner();
+        evm::log(OwnershipTransferStarted {
+            previous_owner: current_owner,
+            new_owner,
+        });
+        Ok(())
+    }
+
+    /// Accepts the ownership of the contract. Can only be called by the
+    /// pending owner.
+    ///
+    /// # Errors
+    ///
+    /// If called by any account other than the pending owner, then the error
+    /// [`OwnableError::UnauthorizedAccount`] is returned.
+    pub fn accept_ownership(&mut self) -> Result<(), Error> {
+        let sender = msg::sender();
+        let pending_owner = self.pending_owner();
+        if sender != pending_owner {
+            return Err(OwnableError::UnauthorizedAccount(
+                OwnableUnauthorizedAccount { account: sender },
+            )
+            .into());
+        }
+        self._transfer_ownership(sender);
+        Ok(())
+    }
+
+    /// Leaves the contract without owner. It will not be possible to call
+    /// [`Self::only_owner`] functions. Can only be called by the current owner.
+    ///
+    /// NOTE: Renouncing ownership will leave the contract without an owner,
+    /// thereby disabling any functionality that is only available to the owner.
+    ///
+    /// # Errors
+    ///
+    /// If not called by the owner, then the error
+    /// [`OwnableError::UnauthorizedAccount`]
```

Comment:
This is copied from Ownable. Is there a way to reference to that documentation as we do in solidiy? e.g. [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/extensions/ERC20Burnable.sol#L20)

Reply:
Maybe this could help https://github.com/OpenZeppelin/rust-contracts-stylus/pull/302#discussion_r1781194065